compiler/src/steps/validate-rest-spec.ts

```diff
@@ -68,6 +68,35 @@ export default async function validateRestSpec (model: model.Model, jsonSpec: Ma
         }
       }
 
+      // are all path parameters properly required or optional?
+      let urlPartsRequired = new Set(urlParts.slice())
```

Comment:
nit: no need to `slice()` to copy the array, as `new Set()` iterates on it, but doesn't modify it.